bin/clang/clangGlobals.h

```diff
@@ -50,10 +57,116 @@ namespace clang {
   class Sema;
 }
 
-namespace sst {
-  extern clang::CompilerInstance *activeCompiler;
-  extern clang::ASTContext *activeASTContext;
-  extern clang::SourceManager *activeSourceManger;
-  extern clang::LangOptions *activeLangOpts;
-  extern clang::Sema *activeSema;
+namespace modes {
+
+enum ModeMask {
+  ENCAPSULATE = 1 << 0,
+  MEMOIZE = 1 << 1,
+  SKELETONIZE = 1 << 2,
+  SHADOWIZE = 1 << 3,
+  PUPPETIZE = 1 << 4
+};
+
+enum Mode {
+  ENCAPSULATE_MODE = 0,
+  MEMOIZE_MODE = 1,
+  SKELETONIZE_MODE = 2,
+  SHADOWIZE_MODE = 3,
+  PUPPETIZE_MODE = 4,
+  NUM_MODES
+};
+
 }
+
+class SkeletonASTVisitor;
+class FirstPassASTVisitor;
+struct SSTPragma;
```

Comment:
struct vs class is annoying in a few places.  Also sst has a lot of mixed struct and class declarations which aren't errors, but aren't great.

Reply:
Agreed. Clean up later.